src/datasets/arrow_dataset.py

```diff
@@ -1815,32 +1825,6 @@ def _map_single(
         if rank is not None and "notebook" in tqdm.__name__:
             print(" ", end="", flush=True)
 
-        # Select the columns (arrow columns) to process
-        if remove_columns is not None and any(col not in self._data.column_names for col in remove_columns):
-            raise ValueError(
-                "Column to remove {} not in the dataset. Current columns in the dataset: {}".format(
-                    list(filter(lambda col: col not in self._data.column_names, remove_columns)),
-                    self._data.column_names,
-                )
-            )
-
-        load_from_cache_file = load_from_cache_file if load_from_cache_file is not None else is_caching_enabled()
-
-        if isinstance(input_columns, str):
-            input_columns = [input_columns]
-
-        if input_columns is not None:
-            for input_column in input_columns:
-                if input_column not in self._data.column_names:
-                    raise ValueError(
-                        "Input column {} not in the dataset. Current columns in the dataset: {}".format(
-                            input_column, self._data.column_names
-                        )
-                    )
-
-        if fn_kwargs is None:
-            fn_kwargs = {}
```

Comment:
This one is a duplicate of the same check we have at lines 1667-1668, so we can remove it.

Reply:
Since the signature of `_map_single` still is `fn_kwargs: Optional[dict] = None`, I would keep this.

Reply:
Oh and I just noticed that since some other conversion were removed from `_map_single`, then we should probably update the type hint of `input_columns` and `remove_columns`. Indeed in `_map_single` they should be `Optional[List[str]]` instead of `Optional[Union[str, List[str]]]`

Reply:
Good catch! Ok, will keep the fn_kwargs check in `_map_single` and update the type hints.